dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java

```diff
@@ -0,0 +1,273 @@
+        for (final Element removedElement : removedElements) {
+           Assert.assertTrue(seekers.contains(removedElement.text()));
+        }
+
+        final Elements addedElements = document.select("span.diff-html-added");
+        for (final Element addedElement : addedElements) {
+            Assert.assertTrue(coneheads.contains(addedElement.text()));
+        }
+    }
+
+
+    /**
+     *Given scenario: We have a page created out of a layout and a container. The Container holds a List of items.
+     * Expected Result:  We create a working copy od the page but this one isn't modified. So whn compared
+     * @throws Exception
+     */
+    @Test
+    public void Test_Page_No_Changes_Expect_Nothing_Changed_Message() throws Exception {
+
+        final String pageName = "seekers-expect-No-difference-page";
+
+        final Contentlet contentlet = new ContentletDataGen(contentGenericId)
+                .languageId(defaultLang.getId())
+                .folder(folder)
+                .host(site)
+                .setProperty("title", "seekers")
+                .setProperty("body", "Skywarp, Starscream, Thundercracker")
+                .nextPersisted();
+
+        contentlet.setIndexPolicy(IndexPolicy.WAIT_FOR);
+        contentlet.setIndexPolicyDependencies(IndexPolicy.WAIT_FOR);
+        contentlet.setBoolProperty(Contentlet.IS_TEST_MODE, true);
+        contentletAPI.publish(contentlet, systemUser, false);
+
+       final HTMLPageAsset pageLive = new HTMLPageDataGen(folder, template).languageId(defaultLang.getId())
+                .pageURL(pageName)
+                .friendlyName(pageName)
+                .title(pageName).nextPersisted();
+
+        final MultiTree multiTreeV1 = new MultiTree(pageLive.getIdentifier(),
+                container.getIdentifier(), contentlet.getIdentifier(),
+                getDotParserContainerUUID(uuid), 0);
+        multiTreeAPI.saveMultiTree(multiTreeV1);
+        HTMLPageDataGen.publish(pageLive);
+
+        //Force a working copy identical to the original
+        final Contentlet workingPage = contentletAPI.checkout(pageLive.getInode(), systemUser, false);
+        contentletAPI.checkin(workingPage, systemUser, false);
+
+        HttpServletRequest request = mock(HttpServletRequest.class);
+        HttpServletResponse response = mock(HttpServletResponse.class);
+        final HttpSession session = mock(HttpSession.class);
+
+        when(request.getRequestURI()).thenReturn("/"+pageName);
+        when(request.getSession()).thenReturn(session);
+        when(request.getSession(false)).thenReturn(session);
+        when(request.getSession(true)).thenReturn(session);
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER_ID))
+                .thenReturn(systemUser.getUserId());
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER)).thenReturn(systemUser);
+        when(session.getAttribute(com.dotmarketing.util.WebKeys.CMS_USER)).thenReturn(systemUser);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'com.dotmarketing.util.WebKeys.CMS_USER' due to existing import 'com.dotmarketing.util.WebKeys'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)